kong/tools/gzip.lua

```diff
@@ -15,7 +16,16 @@ local GZIP_CHUNK_SIZE = 65535
 
 
 local function read_input_buffer(input_buffer)
+  local count = 0
+  local yield_size = GZIP_CHUNK_SIZE * 2
+
   return function(size)
+    count = count + size
+    if count > yield_size then
+      count = 0
+      yield()
+    end
+
     local data = input_buffer:get(size)
```

Comment:
I am thinking whether `input_buffer:get()` itself can yield if the `input_buffer` is set to a socket. If it is a yieldable socket, then the new added `yield()` seems redundant. 

Because according to the `buffer:set()` in https://luajit.org/ext_buffer.html, it supports `string` and `cdata`. Is it possible if the buffer is set to a `cdata` type, which is a socket?

Reply:
Currently (in kong gatway), we are using gzip/gunzip only for string data, as I know no socket to be applied to gzip/gunzip.